drift-transport-netty/src/main/java/io/airlift/drift/transport/netty/server/ThriftServerHandler.java

```diff
@@ -92,6 +98,17 @@ public void channelRead(ChannelHandlerContext context, Object message)
         context.fireChannelRead(message);
     }
 
+    @Override
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause)
```

Comment:
Is this the right place to put this?  The ThriftServerHandler is the inner most handler on the server.  Does it get exceptions from the ChannelHandlers closer to the server socket?

Reply:
I think the chain is next

`LengthFieldBasedFrameDecoder` -> `SimpleFrameCodec` -> `ResponseOrderingHandler` -> `ThriftServerHandler`

If something fails in upstream handlers, the error is propagated all the way downstream to the `ThriftServerHandler` (unless the error is handled somewhere upstream. `SimpleFrameCodec` and `ResponseOrderingHandler` don't have the  `exceptionCaught` method, so the error is propagated all the way to the `ThriftServerHandler`.

Reply:
I'm not sure how the write path works though. What is the propagation chain if the error happens when response is written to the client.

Reply:
I dug into this a bit more.  Uncaught exceptions are propagated down the `ChannelInboundHandler` path.  This handler is the end of this path so this should work.